File: backend/src/main/java/com/primatoos/backend/model/Worker.java
package com.primatoos.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "workers")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Worker {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String function;

    private String phone;

    @Builder.Default
    @Column(nullable = false)
    private boolean active = true;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", unique = true)
    private User user;

    @Setter(AccessLevel.NONE)
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/worker.ts
import type { UserSummaryResponse } from './common'

export interface WorkerResponse {
  id: number
  name: string
  function: string | null
  phone: string | null
  active: boolean
  user: UserSummaryResponse | null
  createdAt: string
}

export interface WorkerSummaryResponse {
  id: number
  name: string
  function: string | null
}

export interface WorkerCreateRequest {
  name: string
  function?: string | null
  phone?: string | null
  userId?: number | null
}

export interface WorkerUpdateRequest {
  name: string
  function?: string | null
  phone?: string | null
  userId?: number | null
}
